kotlin/src/test/kotlin/org/jdbi/v3/core/kotlin/KotlinMapperTest.kt

```diff
@@ -1,169 +1,239 @@
+import org.junit.Rule
 import org.junit.Test
-import java.sql.ResultSet
-import java.sql.ResultSetMetaData
+import org.mockito.junit.MockitoJUnit
+import org.mockito.junit.MockitoRule
 
 class KotlinMapperTest {
+    @Rule
+    @JvmField
+    val mockitoRule: MockitoRule = MockitoJUnit.rule()
 
+    @Rule
+    @JvmField
+    val dbRule: H2DatabaseRule = H2DatabaseRule().withPlugin(KotlinPlugin())
 
-    val resultSet = mock<ResultSet>()
-    val resultSetMetaData = mock<ResultSetMetaData>()
-    val ctx = StatementContextAccess.createContext(HandleAccess.createHandle())
-
-    private data class DataClassWithOnlyPrimaryConstructor(val id: Int, val name: String)
+    private lateinit var handle: Handle
 
     @Before
-    fun setUpMocks() {
-        whenever(resultSet.metaData).thenReturn(resultSetMetaData)
+    fun setup() {
+        handle = dbRule.sharedHandle
+        handle.execute("CREATE TABLE the_things(id integer, first text)")
+        handle.execute("CREATE TABLE the_other_things(id integer, other text)")
     }
 
-    @Test fun testDataClassWithOnlyPrimaryConstructor() {
+    data class DataClassWithOnlyPrimaryConstructor(val id: Int, val first: String)
 
-        val mapper = KotlinMapper(DataClassWithOnlyPrimaryConstructor::class.java)
-        mockColumns("id", "name")
+    @Test
+    fun testDataClassWithOnlyPrimaryConstructor() {
+        val expected = DataClassWithOnlyPrimaryConstructor(1, "does this work?")
 
-        whenever(resultSet.getInt(1)).thenReturn(1)
-        whenever(resultSet.getString(2)).thenReturn("one")
-        whenever(resultSet.wasNull()).thenReturn(false)
+        handle.createUpdate("INSERT INTO the_things(id, first) VALUES(:id, :first)")
+                .bindBean(expected)
+                .execute()
 
-        val thing = mapper.map(resultSet, ctx)
-
-        assertThat(thing).isEqualToComparingFieldByField(DataClassWithOnlyPrimaryConstructor(1, "one"))
+        val result = handle.createQuery("SELECT * from the_things")
+                .mapTo<DataClassWithOnlyPrimaryConstructor>()
+                .single()
 
+        assertThat(result).isEqualTo(expected)
     }
 
-    @Test fun shouldUseSecondOccurrenceOfColumnWhenMultipleColumnsWithSameNameArePresent() {
-
-        val mapper = KotlinMapper(DataClassWithOnlyPrimaryConstructor::class.java)
-        mockColumns("id", "name", "name")
-
-        whenever(resultSet.getInt(1)).thenReturn(1)
-        whenever(resultSet.getString(2)).thenReturn("one")
-        whenever(resultSet.getString(3)).thenReturn("two")
-        whenever(resultSet.wasNull()).thenReturn(false)
-
-        val thing = mapper.map(resultSet, ctx)
-
-        assertThat(thing).isEqualToComparingFieldByField(DataClassWithOnlyPrimaryConstructor(1, "two"))
-
+    @Test
+    fun shouldUseSecondOccurrenceOfColumnWhenMultipleColumnsWithSameNameArePresent() {
+        TODO("Pretty sure this test isn't necessary anymore since we're using regular 'findColumnIndex', please confirm")
```

Comment:
I concur.

Reply:
Alright, I'll remove the tests (there's another in KotlinPluginTest if I remember correctly).